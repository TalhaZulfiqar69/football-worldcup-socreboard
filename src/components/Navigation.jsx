import {
  Box,
  Flex,
  HStack,
  useColorModeValue,
  Heading,
} from "@chakra-ui/react";

const Navigation = () => {
  return (
    <>
      <Box
        bg={useColorModeValue("gray.100", "gray.900")}
        px={4}
        bgGradient="linear(to-l, #7928CA, #FF0080)"
      >
        <Flex h={16} alignItems={"center"} justifyContent={"space-between"}>
          <HStack spacing={8} alignItems={"center"}>
            <Heading as="h3" size="md" color={"#ffffff"}>
              Live Football World Cup Scoreboard
            </Heading>
          </HStack>
          <Flex alignItems={"center"}></Flex>
        </Flex>
      </Box>
    </>
  );
};

export default Navigation;
