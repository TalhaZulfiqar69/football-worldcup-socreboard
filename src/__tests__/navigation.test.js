import { render, screen } from "@testing-library/react";
import { axe } from "jest-axe";
import "jest-axe/extend-expect";
import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import Navigation from "../components/Navigation";
import mockTheme from "../chakra-ui-mock-theme";

const extendedTheme = extendTheme({ ...mockTheme });

// TESTCASE # 1: Renders without crashing
// DESCRIPTION: Here I am testing if the component renders without any errors correctly or not.
test("renders without crashing", () => {
  render(
    <ChakraProvider theme={extendedTheme}>
      <Navigation />
    </ChakraProvider>
  );
});

// TESTCASE # 2: Component has no accessibility violations
// DESCRIPTION:  Here I am testing the component for accessibility issues using tools like Jest Axe or React Testing Library's axe function.
test("component has no accessibility violations", async () => {
  const { container } = render(
    <ChakraProvider theme={extendedTheme}>
      <Navigation />
    </ChakraProvider>
  );
  const results = await axe(container);

  expect(results).toHaveNoViolations();
});

// TESTCASE # 3: Displays correct heading text
// DESCRIPTION: Here I am testing if the component displaying the correct heading text as expected or not.
test("displays correct heading text", () => {
  render(
    <ChakraProvider theme={extendedTheme}>
      <Navigation />
    </ChakraProvider>
  );
  const headingElement = screen.getByText("Live Football World Cup Scoreboard");
  expect(headingElement).toBeInTheDocument();
});

// TESTCASE # 4: Matches snapshot
// DESCRIPTION:  Here I am testing that if something change in component each time when testcases executes.
test("matches snapshot", () => {
  const { asFragment } = render(
    <ChakraProvider theme={extendedTheme}>
      <Navigation />
    </ChakraProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
