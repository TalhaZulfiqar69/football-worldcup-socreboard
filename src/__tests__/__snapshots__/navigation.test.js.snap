// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-ug4a5h"
  >
    <div
      class="css-19pkskf"
    >
      <div
        class="chakra-stack css-pa7cn"
      >
        <h3
          class="chakra-heading css-k154rz"
        >
          Live Football World Cup Scoreboard
        </h3>
      </div>
      <div
        class="css-70qvj9"
      />
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
